src/fpm_model.f90

```diff
@@ -86,6 +87,15 @@ module fpm_model
     !> Native libraries to link against
     type(string_t), allocatable :: link_libraries(:)
 
+    !> Fortran compiler flags
+    character(len=:), allocatable :: flags
+
+    !> C compiler flags
+    character(len=:), allocatable :: c_flags
```

Comment:
We don't need both `flags` and `c_flags` for source files, just `flags` which is set depending on what type of source file it is. This will simplifies logic later on.